daweb/js2/sablony-komponenty/sablonovaci-systemy.md

```diff
@@ -0,0 +1,83 @@
+## Šablonovací systémy
+
+Jedním z najčastějších úkonů při tvorbě webových stránek je vytváření obsahu stránky z nějakých dat. My k tomuto účelu zatím používáme vlastnost `innerHTML`, což znamená, že musíme vždy sestavit řetězec obsahující přesně takové HTML, jaké poté chceme vložit do stránky. Tento postup nám při tvorbě větších stránek brzy přeroste přes hlavu. Z tohoto důvodu se většina vývojářů obrací k nějakému šablonovacímu systému, který tvorbu HTML v JavaScirptu usnadní.
+
+Takových systému je pro JavaScript k dispozici nepřeberné množství. Tady je seznam jen několika nejznámějších:
+
+- [Handlebars](https://handlebarsjs.com/)
+- [EJS](https://ejs.co/)
+- [Pug](https://pugjs.org/api/getting-started.html)
+- [Nunjucks](https://mozilla.github.io/nunjucks/)
+- [uhtml](https://github.com/WebReflection/uhtml)
+
+S příchodem frameworku React, ke kterému zde postupně s napětím směřujeme, se stal populární šablonovací systém [JSX](https://reactjs.org/docs/introducing-jsx.html), který je součástí frameworku React. My proto JSX v tomto kurzu budeme také používat, protože se tak mimochodem naučíme i základy Reactu ještě dříve, než se k němu oficiálně prokoušeme.
+
+## Základy JSX
+
+Abychom mohli pracovat s JSX, potřebujeme mít správně nastavený projekt. V úplně čístém JavaScriptu by nám JSX šablony nefungovaly. Naštěstí náš startovač projektů už je na toto připraven. Stačí tedy založi vanilla projekt už známým příkazem
+
+```sh
+npm init kodim-app@latest muj-projekt vanilla
```

Comment:
Říkat jsx vanilla mi přijde hodně matoucí.

Reply:
Máš pravdu, předělal jsem ten kit, aby se to jmenovalo jsx.